plugins/scale/scale-title-overlay.cpp

```diff
@@ -0,0 +1,326 @@
+                            wf::scale_transformer& tr,
+                            nonstd::observer_ptr<wf::custom_data_t> data)
+    {
+        wayfire_view view = tr.get_transformed_view();
+        view_title_overlay *overlay = dynamic_cast<view_title_overlay*>(data.get());
+
+        if ((show_view_title_overlay == title_overlay_t::NEVER) ||
+            ((show_view_title_overlay == title_overlay_t::MOUSE) &&
+             (view != last_title_overlay)))
+        {
+            return;
+        }
+
+        wf::geometry_t geometry;
+        GLuint tex = overlay->get_texture(fb.scale, geometry);
+
+        if (tex == (GLuint) - 1)
+        {
+            return;
+        }
+
+        auto ortho = fb.get_orthographic_projection();
+        OpenGL::render_begin(fb);
+        for (const auto& box : damage)
+        {
+            fb.logic_scissor(wlr_box_from_pixman_box(box));
+            OpenGL::render_transformed_texture(tex, geometry, ortho,
+                {1.0f, 1.0f, 1.0f, tr.alpha}, OpenGL::TEXTURE_TRANSFORM_INVERT_Y);
+        }
+
+        OpenGL::render_end();
+    }
+}
+
+{}
+
+void scale_show_title::init(wf::output_t *output)
+{
+    this->output = output;
+    output->connect_signal("scale-filter", &view_filter);
+    output->connect_signal("scale-padding", &scale_padding);
+    output->connect_signal("scale-transformer-added", &add_title_overlay);
+    output->connect_signal("scale-end", &scale_end);
+}
+
+void scale_show_title::fini()
+{
+    output->disconnect_signal(&view_filter);
+    output->disconnect_signal(&add_title_overlay);
+    output->disconnect_signal(&scale_end);
+    if (mouse_update_connected)
+    {
+        wf::get_core().disconnect_signal(&mouse_update);
+        mouse_update_connected = false;
+    }
+}
+
+void scale_show_title::update_title_overlay_opt()
+{
+    const std::string& tmp = show_view_title_overlay_opt;
+    if (tmp == "all")
+    {
+        show_view_title_overlay = title_overlay_t::ALL;
+    } else if (tmp == "mouse")
+    {
+        show_view_title_overlay = title_overlay_t::MOUSE;
+    } else
+    {
+        show_view_title_overlay = title_overlay_t::NEVER;
+    }
+
+    if (show_view_title_overlay == title_overlay_t::MOUSE)
+    {
+        update_title_overlay_mouse();
+        if (!mouse_update_connected)
+        {
+            wf::get_core().connect_signal("pointer_motion_post",
+                &mouse_update);
+            mouse_update_connected = true;
+        }
+    }
+}
+
+void scale_show_title::update_title_overlay_mouse()
+{
+    wayfire_view v;
+
+    /* TODO: better way to determine if scale's grab interface is active! */
+    wf::option_wrapper_t<bool> interact{"scale/interact"};
+
+    if (interact)
+    {
+        /* we can use normal focus tracking */
+        v = wf::get_core().get_cursor_focus_view();
+    } else
+    {
+        auto& core = wf::get_core();
+        v = core.get_view_at(core.get_cursor_position());
+    }
+
+    if (v)
```

Comment:
Hmm, if we have a dialog on top of a view, should we not display the title on the frontmost dialog (not on the main view)? The main view will not be visible for the most part.

Reply:
Modal dialogs e.g of dconf-editor commonly use empty titles.

Reply:
@damianatorrpm no, I think you missed my point. The current approach does this:

- render the base view
- render the base view's title
- render the dialog

The dialog is drawn over the title, so no title is visible at all...

Reply:
I haven't really thought of dialogs, but it is true that they will occlude the parent. One solution could be to show the overlay for both parent and dialog with the exception that we don't show a dialog's title if it is empty. This is mainly a problem if the overlay is centered -- if it is on top or bottom, it will still be visible.